### PR TITLE
Exception handling

### PR DESCRIPTION
**Pull request by ystankevych**

### CODE REVIEW

File: src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler {
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java]:
```suggestion
```

File: src/main/java/mate/academy/exception/errors/BookApiError.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package mate.academy.exception.errors;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import java.time.LocalDateTime;
+import org.springframework.http.HttpStatus;
+
+public record BookApiError(
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/exception/errors/BookApiError.java]:
```suggestion
public record BookApiException(
```

File: src/main/java/mate/academy/exception/errors/BookApiError.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package mate.academy.exception.errors;
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/exception/errors/BookApiError.java]: move it to exception package and remove errors

File: src/main/java/mate/academy/service/BookServiceImpl.java [hunk 4]
```diff
@@ -45,12 +45,16 @@ public List<BookDto> findAll() {
...
     }
 
     @Override
     public void deleteById(Long id) {
-        bookRepository.deleteById(id);
+        if (bookRepository.existsById(id)) {
+            bookRepository.deleteById(id);
+        } else {
+            throw new EntityNotFoundException(getNotFoundMessage(id));
+        }
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/service/BookServiceImpl.java]: for now, I prefer not to complicate your logic with additional checking

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
good job! Let’s  improve your solution ;)

File: src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+@ControllerAdvice
+public class CustomGlobalExceptionHandler {
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    protected ResponseEntity<Object> invalidArgument(MethodArgumentNotValidException ex) {
+        return buildResponseEntity(new BookApiException(BAD_REQUEST,
+                LocalDateTime.now(), getErrorsMessage(ex.getBindingResult())));
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java]:
```suggestion
    protected ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException ex) {
```

File: src/main/java/mate/academy/controller/BookController.java [hunk 6]
```diff
@@ -41,6 +46,7 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/BookController.java]: If you added response status for one endpoint, let's add for others

**Review by Elena-Bruyako — APPROVED** [review]
Good job!

**Pull request merged**